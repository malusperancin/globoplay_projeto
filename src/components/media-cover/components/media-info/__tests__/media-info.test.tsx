/* eslint-disable testing-library/prefer-screen-queries */
import { render } from "@testing-library/react";
import MediaInfo from "../media-info";
import MediaInfoType from "../media-info-type";

const mock: MediaInfoType = {
  card: {
    backgroundUrl:
      "http://s2.glbimg.com/CTUVKaH8UAyDzA6yNDZWmVAADKc=/1280x720/filters:quality(40)/https://s2.glbimg.com/9lSn3h-fSHgG2sX9PJaSx1dp3dI=/https://i.s3.glbimg.com/v1/AUTH_c3c606ff68e7478091d1ca496f9c5625/internal_photos/bs/2022/A/B/w3HJv6R32ElJ3lDkAGHg/2022-2881-sob-pressao-t5-destaque-foco-direita.jpg",
    description:
      "Novos episódios disponíveis! Na 5ª temporada, a saúde mental e física da equipe médica preocupa",
    hasVideo: false,
    broadcast: false,
    headline: "Sob Pressão",
    headlineImage:
      "http://s2.glbimg.com/LrVLlrMezNJ026waR0sgyQVQ13o=/0x110/https://s2.glbimg.com/uB9M8WyLrkLworWdYQFmF7MJ52s=/fit-in/0x364/filters:fill(transparent)/https://i.s3.glbimg.com/v1/AUTH_c3c606ff68e7478091d1ca496f9c5625/internal_photos/bs/2022/N/3/KBN5XOSiAIl4MvuABTdg/2022-2881-sob-pressao-t5-logo-destaque.png",
    idSalePage: "globoplay-web",
    logoUrl:
      "http://s2.glbimg.com/LrVLlrMezNJ026waR0sgyQVQ13o=/0x110/https://s2.glbimg.com/uB9M8WyLrkLworWdYQFmF7MJ52s=/fit-in/0x364/filters:fill(transparent)/https://i.s3.glbimg.com/v1/AUTH_c3c606ff68e7478091d1ca496f9c5625/internal_photos/bs/2022/N/3/KBN5XOSiAIl4MvuABTdg/2022-2881-sob-pressao-t5-logo-destaque.png",
    mediaId: false,
    position: 1,
    serviceId: 151,
    titleId: "PhQ7Wtzn6C",
    titleType: "SERIE",
    type: "TITLE",
  },
  contentType: "OFFERHIGHLIGHT",
};

const mediaInfoHighlight = <MediaInfo isVisible={true} info={mock} />;

describe("Media Info", () => {
  it("should render without throwing an error", () => {
    const { container } = render(mediaInfoHighlight);

    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it("should render a Highlight Media Info", () => {
    const { getByTestId } = render(mediaInfoHighlight);
    const mediaInfo = getByTestId("media-info-highlight");
    expect(mediaInfo).toBeTruthy();
  });
});
