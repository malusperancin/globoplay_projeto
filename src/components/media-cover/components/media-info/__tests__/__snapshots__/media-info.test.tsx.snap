// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Info should render without throwing an error 1`] = `
<div>
  <div
    class="offerHighlight"
    data-testid="media-info-highlight"
  >
    <img
      alt=""
      class="logo"
      src="http://s2.glbimg.com/LrVLlrMezNJ026waR0sgyQVQ13o=/0x110/https://s2.glbimg.com/uB9M8WyLrkLworWdYQFmF7MJ52s=/fit-in/0x364/filters:fill(transparent)/https://i.s3.glbimg.com/v1/AUTH_c3c606ff68e7478091d1ca496f9c5625/internal_photos/bs/2022/N/3/KBN5XOSiAIl4MvuABTdg/2022-2881-sob-pressao-t5-logo-destaque.png"
    />
    <p
      class="title"
    >
      Novos episódios disponíveis! Na 5ª temporada, a saúde mental e física da equipe médica preocupa
    </p>
  </div>
</div>
`;
